### PR TITLE
Indicate file compression was skipped

### LINKED ISSUE #245: Indication that file compression was reverted
Follow-up to #229. Maybe this could be a toast? Another thing to consider is, should this happen only when Curtail manually reverts the file, or also when the compressors skip the compression? The latter might be harder to figure out.

### PR DESCRIPTION
**Pull request by ARAKHN1D**
Adds a visual indication that file compression was skipped, along with a button explaining why.

The skipping mechanism was also improved so it actually skips files in overwrite mode rather than just reverting them.

indicator

Supersedes #249 because I did something weird with Git on accident :D
Closes #245

### CODE REVIEW

File: src/compressor.py [hunk 1]
```diff
@@ -107,16 +113,27 @@ def run_command(self, command, result_item):
...
+                                result_item.skipped = True
+
+                            # Remove temporary file
+                            Path(result_item.filename).unlink(True)
+                            result_item.filename = result_item.original_filename
+                            result_item.new_filename = result_item.original_filename
+                            new_file_data = Path(result_item.new_filename)
+                            result_item.new_size = new_file_data.stat().st_size
+                    else:
+                        if filecmp.cmp(result_item.filename, result_item.new_filename):
```
Comment (Huluti) [inline src/compressor.py:135]: What does the filecmp.cmp call check please here?
It's to see if something has been done on the file? Better than checking size right?

Comment (Huluti) [reply src/compressor.py:135]: Maybe else/if can be combined in an elif?

**Comment by Huluti** [comment]
The look is perfect for me! Will try to test your pr soon to understand all and validate changes :)

**Comment by Huluti** [comment]
Also, please add you in the contributors of the app in the About window, you deserve it!

File: src/compressor.py (hunk 1 above)
Comment (ARAKHN1D) [reply src/compressor.py:135]: The filecmp.cmp check compares the data returned by the os.stat function, which checks a bunch of different file properties, including file size. However, in hindsight I don't think it's very necessary to have to check all of that. I think I was just being overly cautious, and it's likely not worth importing an entire module when the file size can just be compared. 

Good catch on the else/if. I'll fix that in another commit.

**Comment by Huluti** [comment]
Just have rebased the PR @ARAKHN1D 
Could you add more comments, especially in the “# Manually skip files if necessary” section because the logic is starting to get more complicated.
Thank you very much!

**Comment by Huluti** [comment]
Maybe we could even create a document that explain the process and what we want for the different formats (png, jpg, webp, svg) and the modes (safe, non-safe).

**Comment by ARAKHN1D** [comment]
Yes. I'll add more comments, and if that's not enough I'll create a document.

**Comment by Huluti** [comment]
Thanks a lot. Will do a final check this week to finally merge it.

**Pull request merged**